File: backend/internal/vo/course.go
package vo

import "time"

type CourseVO struct {
	ID                uint64    `json:"id"`
	CourseCode        string    `json:"courseCode"`
	CourseName        string    `json:"courseName"`
	CourseDescription string    `json:"courseDescription"`
	OwnerUserID       uint64    `json:"ownerUserId"`
	Status            string    `json:"status"`
	CreatedAt         time.Time `json:"createdAt"`
	UpdatedAt         time.Time `json:"updatedAt"`
	MyRole            string    `json:"myRole,omitempty"`
}

type CourseMemberVO struct {
	ID         uint64    `json:"id"`
	CourseID   uint64    `json:"courseId"`
	UserID     uint64    `json:"userId"`
	Role       string    `json:"role"`
	JoinStatus string    `json:"joinStatus"`
	JoinedAt   time.Time `json:"joinedAt"`
}

File: frontend/src/types/course.ts
export type CourseRole = 'owner' | 'teacher' | 'student'
export type CourseStatus = 'active' | 'archived' | 'deleted'
export type CourseJoinStatus = 'active' | 'removed'

export interface CourseVO {
  id: number
  courseCode: string
  courseName: string
  courseDescription: string
  ownerUserId: number
  status: CourseStatus
  createdAt: string
  updatedAt: string
  myRole?: CourseRole
}

export interface CourseMemberVO {
  id: number
  courseId: number
  userId: number
  username: string
  role: CourseRole
  joinStatus: CourseJoinStatus
  joinedAt: string
}

export interface CreateCoursePayload {
  courseCode: string
  courseName: string
  courseDescription: string
}

export interface UpdateCoursePayload {
  courseName: string
  courseDescription: string
}

export interface AddCourseMemberPayload {
  username: string
  role: Exclude<CourseRole, 'owner'>
}

export interface UpdateCourseMemberPayload {
  role: Exclude<CourseRole, 'owner'>
}
